### PR TITLE
Unity

### PR DESCRIPTION
**Pull request by MiLk**
- Use the `__cplusplus`define to disable C++11 features if they are not available.
- New `UnityRelease` target in the XCode project, using `gnu++98` and `libstdc++` instead of `gnu++0x` and `libc++`.
- Use `libcurl.a` in addition of `libcurl_arm64.a` to make the sdk available for `armv7` and `armv7s`.
- Update the Makefile.

### CODE REVIEW

**Comment by MiLk** [comment]
I would like to merge #28 before this one, as it will require extra work to disable the new feature in Unity.

**Comment by unity3diy** [comment]
fine :+1:

File: src/rpc_stream.cpp [hunk 1]
```diff
@@ -201,7 +222,9 @@ namespace mage {
 	}
 
 	std::string RPC::GetMsgStreamUrl(Transport transport) const {
+#ifndef UNITY
```
Comment (ronkorving) [inline src/rpc_stream.cpp:225]: Does this imply the Unity implementation is not thread safe?

Comment (MiLk) [inline src/rpc_stream.cpp:225]: Yes, but we should not use threads with unity. (JNI can't be called from another thread).

Comment (ronkorving) [inline src/rpc_stream.cpp:225]: Okay :) (sounds like a horrible limitation, but so be it)

**Comment by MiLk** [comment]
No longer relevant.
As it doesn't work with Unity 5.
And it's much more cleaner to do it in C#.

**Pull request closed without merging**